### ISSUE TITLE
Temporary disruptions to state store make teraslice unresponsive

### ISSUE THREAD

**Issue by erik-stephens** [terascope/teraslice#1109]
We're required to restart the master to recover, which isn't so bad since doesn't disrupt jobs.  Plus, adding a liveness probe to the pod might be a workaround to get us by for a while.

These are the only msgs we see:
```
15:05:05.360Z ERROR cluster_master: invoking elasticsearch-api client->search(), caused by Error: Request Timeout after 120000ms (module=api_service, err.code=INTERNAL_SERVER_ERROR)
    TSError: invoking elasticsearch-api client->search(), caused by Error: Request Timeout after 120000ms
        at _errorHandlerFn (/app/source/packages/elasticsearch-api/index.js:531:24)
        at tryCatcher (/app/source/node_modules/bluebird/js/release/util.js:16:23)
        at Promise._settlePromiseFromHandler (/app/source/node_modules/bluebird/js/release/promise.js:512:31)
        at Promise._settlePromise (/app/source/node_modules/bluebird/js/release/promise.js:569:18)
        at Promise._settlePromise0 (/app/source/node_modules/bluebird/js/release/promise.js:614:10)
        at Promise._settlePromises (/app/source/node_modules/bluebird/js/release/promise.js:690:18)
        at _drainQueueStep (/app/source/node_modules/bluebird/js/release/async.js:138:12)
        at _drainQueue (/app/source/node_modules/bluebird/js/release/async.js:131:9)
        at Async._drainQueues (/app/source/node_modules/bluebird/js/release/async.js:147:5)
        at Immediate.Async.drainQueues [as _onImmediate] (/app/source/node_modules/bluebird/js/release/async.js:17:14)
        at runCallback (timers.js:705:18)
        at tryOnImmediate (timers.js:676:5)
        at processImmediate (timers.js:658:5)
        caused by, Error: Request Timeout after 120000ms
        at /app/source/node_modules/elasticsearch/src/lib/transport.js:355:15
        at Timeout.<anonymous> (/app/source/node_modules/elasticsearch/src/lib/transport.js:384:7)
        at ontimeout (timers.js:436:11)
        at tryOnTimeout (timers.js:300:5)
        at listOnTimeout (timers.js:263:5)
        at Timer.processTimers (timers.js:223:10)
    Caused by: Error: Request Timeout after 120000ms
        at /app/source/node_modules/elasticsearch/src/lib/transport.js:355:15
        at Timeout.<anonymous> (/app/source/node_modules/elasticsearch/src/lib/transport.js:384:7)
        at ontimeout (timers.js:436:11)
        at tryOnTimeout (timers.js:300:5)
        at listOnTimeout (timers.js:263:5)
        at Timer.processTimers (timers.js:223:10)
```

**Comment by erik-stephens** [comment]
Forgot to mention that I think only a subset of endpoints stay unresponsive.  Only the assets related endpoints come to mind.

**Comment by godber** [comment]
I tried to reproduce this in the simple native clustering case and was unable to.  So the issue is not as simple as, stopping ES and then it hangs.  I tried the following scenarios:

* native clustering, with no jobs running, stop ES, verify API issue, start ES, all API endpoints, including assets work
* native clustering, with jobs running, stop ES, verify API issue, start ES, all API endpoints, including assets work

Next steps would be to repeat these tests in minikube or a test cluster.

If we see this again in production let me know so I can look at the master.
